ire/src/main/scala/hu/bme/mit/ire/messages/Messages.scala

```diff
@@ -9,7 +9,23 @@ import scala.collection.mutable
 
 class ReteMessage() {}
 
-case class ChangeSet(positive: TupleBag = Vector(), negative: TupleBag = Vector()) extends ReteMessage
+trait DataMessage extends ReteMessage {
+  def changeSets: Seq[TupleBag]
+  def createNew(bags: Seq[TupleBag]): DataMessage
```

Comment:
Let's write a map function that forwards the parameter function to all the bags.